helix-view/src/document.rs

```diff
@@ -775,24 +919,34 @@ impl Document {
     // transact(Fn) ?
 
     // -- LSP methods
-
+    #[inline]
     pub fn identifier(&self) -> lsp::TextDocumentIdentifier {
         lsp::TextDocumentIdentifier::new(self.url().unwrap())
     }
 
+    #[inline]
     pub fn versioned_identifier(&self) -> lsp::VersionedTextDocumentIdentifier {
         lsp::VersionedTextDocumentIdentifier::new(self.url().unwrap(), self.version)
     }
 
```

Comment:
Note: it's usually recommended to leave these methods as is and only mark them inline if you can prove that rust doesn't already inline them. In most cases LLVM will pick these up. It's fine to keep these though

Reply:
That's not *quite* true.  Within the same crate, they'll be inlined as needed regardless, yes.  But cross-crate I believe the attribute is required to allow inlining in other crates at all.  At least, without link-time optimization enabled.  See [the perf book](https://nnethercote.github.io/perf-book/inlining.html).

I ran into this when working on my pathtracing renderer, when I split my linear algebra code out into a separate crate.  After doing that, the whole renderer dropped in performance (I think by over 2x, IIRC) because literally none of those hot functions were being inlined anymore.  Marking them `#[inline]` fixed the issue.

Having said that, I doubt these functions are hot.  So I agree the `#[inline]` is unnecessary.  When writing my own code, I usually think of `#[inline]` as roughly synonymous with "I believe (or have shown through profiling) that this function is hot".  And things usually work out well that way.

Reply:
I instead `inline`d all functions that were referenced 6+ times in total and was called outside of crate boundaries. I think two functions that weren't `inline`d were referenced 100+ times.